### PR TITLE
MLE-13685 Added support for patch builder methods

### PR DESCRIPTION
**Pull request by rjrudin**
All of `PatchBuilder` and its impl was manually added. Was able to code generate the methods in the wrong place - PlanBuilder - and then move them over, and then manually implement them.

### CODE REVIEW

File: marklogic-client-api/src/main/java/com/marklogic/client/expression/PlanBuilder.java [hunk 1]
```diff
@@ -227,6 +227,58 @@ protected PlanBuilder(
...
+  /**
+  * Add an error-handler to the Optic Pipeline to catch Optic Update runtime errors. The runtime errors are added in the errors column. If no error occurred the value of the error column is null. When added, the error-handler should be the last operator before op:result.
+  * @param action  The Optic Plan. You can either use the XQuery =&gt; chaining operator or specify the variable that captures the return value from the previous operation.
+  * @return  a ModifyPlan object
+  */
+  public abstract ModifyPlan onError(XsStringVal action);
+  /**
+  * Add an error-handler to the Optic Pipeline to catch Optic Update runtime errors. The runtime errors are added in the errors column. If no error occurred the value of the error column is null. When added, the error-handler should be the last operator before op:result.
+  * @param action  The Optic Plan. You can either use the XQuery =&gt; chaining operator or specify the variable that captures the return value from the previous operation.
+  * @param errorColumn  Valid options are: "fail" - stop procesisng and "continue" - add an error to the error column and continue processing. See {@link PlanBuilder#col(XsStringVal)}
```
Comment (BillFarber) [inline marklogic-client-api/src/main/java/com/marklogic/client/expression/PlanBuilder.java:244]: minor typo in processing

File: marklogic-client-api/src/main/java/com/marklogic/client/expression/PlanBuilder.java [hunk 2]
```diff
@@ -227,6 +227,58 @@ protected PlanBuilder(
...
+  * @return  a ModifyPlan object
+  */
+  public abstract ModifyPlan onError(String action, String errorColumn);
+  /**
+  * Add an error-handler to the Optic Pipeline to catch Optic Update runtime errors. The runtime errors are added in the errors column. If no error occurred the value of the error column is null. When added, the error-handler should be the last operator before op:result.
+  * @param action  The Optic Plan. You can either use the XQuery =&gt; chaining operator or specify the variable that captures the return value from the previous operation.
+  * @param errorColumn  Valid options are: "fail" - stop procesisng and "continue" - add an error to the error column and continue processing. See {@link PlanBuilder#col(XsStringVal)}
```
Comment (BillFarber) [inline marklogic-client-api/src/main/java/com/marklogic/client/expression/PlanBuilder.java:251]: minor typo in processing

**Review by BillFarber — APPROVED** [review]

File: marklogic-client-api/src/main/java/com/marklogic/client/expression/PlanBuilder.java (hunk 1 above)
Comment (rjrudin) [reply marklogic-client-api/src/main/java/com/marklogic/client/expression/PlanBuilder.java:244]: Created MLE-13700 to fix this.

File: marklogic-client-api/src/main/java/com/marklogic/client/expression/PlanBuilder.java (hunk 2 above)
Comment (rjrudin) [reply marklogic-client-api/src/main/java/com/marklogic/client/expression/PlanBuilder.java:251]: Created MLE-13700 to fix this.

**Pull request merged**